lib/utils/location_data.dart

```diff
@@ -0,0 +1,39 @@
+import 'package:http/http.dart';
+import 'package:intl/intl.dart';
+import 'dart:convert';
+
+class LocationData {
+  LocationData({required this.location, required this.url});
+
+  String location;
+  String url;
+  String? time;
+  int? dayOfWeek;
+  int? dayOfyear;
+  int? unixTime;
+  int? weekNumber;
+  bool? isDayTime;
```

Comment:
Anybody, who obtains a `LocationData` object, can currently write to these fields, which is probably not desirable. Can you refactor this, so outsiders can only read these fields and only `fetchData` can modify them?